### DISCUSSION TITLE
Web Component 隔离了 css，UI 框架如何让使用者可以自定义响应式布局的 breakpoint？

### DISCUSSION THREAD

**Discussion by mwxxhdb** [ofajs/discussion#7] [General]
UI 框架通常需要使用 media 查询决定响应式布局，如何让响应式布局的 breakpoint (示例中为600px)可以自定义？
```css
@media screen and (max-width: 600px) {
}
```

**Comment by kirakiray** [comment]
抱歉，当前 ofa.js的版本可能没有什么好的写法支持这种需求，要监听变量值去修改组件内的style内容，具体如下：

```html
<template page>
  <style :html="innerStyle"></style>
  <script>
    export default {
      data: {
        swidth: 600, // 监听变化
        innerStyle: "",
      },
      watch: {
        swidth(swidth) {
          this.innerStyle = `
          @media screen and (max-width: ${swidth}px) {
            xxx
          }
          `;
        },
      },
    };
  </script>
</template>
```

不过满足你这个需求的特征还在讨论中，ofa.js 会在下一个版本，有更好的写法支持你这个需求。

**Comment by kirakiray** [comment]
样式内支持渲染数据的 feature issuse 在：https://github.com/kirakiray/Xhear/issues/110

**Comment by kirakiray** [comment]
已更新到 [4.4.10](https://github.com/kirakiray/ofa.js/releases/tag/4.4.10)，请更新你的cdn地址：

```html
<script src="https://cdn.jsdelivr.net/gh/kirakiray/ofa.js@4.4.10/dist/ofa.min.js"></script>
```

4.4.10 开始，组件或页面模块内的style元素，支持 `data()` 函数：

```html
<template component>
<style>
  @media screen and (max-width: data(mwidth)) {
    ...
  }
</style>
<script>
export default {
  data:{
    mwidth:'1280px',
  }
};
</script>
</template>
```

你可以查看这个自动化测试案例的[源代码](https://github.com/kirakiray/ofa.js/blob/main/test/cases/style-data/sd-demo.html)，[点击预览](https://kirakiray.github.io/ofa.js/test/cases/style-data/test.html)
